build/gen_stub.php

```diff
@@ -574,7 +582,7 @@ public static function fromString(string $typeString): self {
     /**
      * @param SimpleType[] $types
      */
-    private function __construct(array $types) {
+    private function __construct(array $types, public readonly bool $isIntersection = false) {
```

Comment:
PHP 7.1 compatible syntax should be used in gen_stub.php because it's part of the build system and we aim to be compatible with this version. Although, I think we could update to PHP 7.4 sometime soon :)

Reply:
Nit: I'd be happier if `$isIntersection` wasn't an optional parameter so that it would be more explicit whether a union or an intersection type is instantiated.